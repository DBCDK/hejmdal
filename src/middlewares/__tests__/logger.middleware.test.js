import {assert} from 'chai';
import sinon from 'sinon';

import {LoggerMiddleware} from '../logger.middleware';
import {log} from '../../utils/logging.util';

describe('LoggerMiddleware tests', () => {
  let sandbox;

  beforeEach(() => {
    sandbox = sinon.sandbox.create();
  });

  afterEach(() => {
    sandbox.restore();
  });

  it('Should invoke next and make call to log.info', () => {
    const spy = sandbox.spy(log, 'info');
    const next = sandbox.mock();
    const ctx = {
      request: {
        method: 'method',
        url: 'url',
        header: 'header'
      },
      response: {
        status: 'status',
        message: 'message'
      }
    };

    const res = LoggerMiddleware(ctx, next);

    return res.then(() => {
      assert.isTrue(next.called, 'next was invoked');
      assert.isTrue(spy.called, 'log.info was invoked');
      assert.equal(spy.args[0][0], 'page request');
    });
  });

  it('Should invoke next and make call to log.error', () => {
    const spy = sandbox.spy(log, 'error');
    const next = sandbox.mock();
    const ctx = {};

    const res = LoggerMiddleware(ctx, next);

    return res.then(() => {
      assert.isTrue(next.called, 'next was invoked');
      assert.isTrue(spy.called, 'log.error was invoked');
      assert.equal(spy.args[0][0], 'parsing of ctx object failed');
    });
  });
});
