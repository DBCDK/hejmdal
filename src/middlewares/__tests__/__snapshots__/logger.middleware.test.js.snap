// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggerMiddleware tests Should invoke next and make call to log.error 1`] = `
[
  "parsing of request failed",
  {
    "error": [TypeError: Cannot read property 'baseUrl' of null],
    "req": null,
  },
]
`;

exports[`LoggerMiddleware tests Should invoke next and make call to log.info 1`] = `
[MockFunction] {
  "calls": [
    [
      "page request",
      {
        "baseUrl": "url/path",
        "clientId": "",
        "requestObject": {
          "header": "header",
          "method": "method",
          "params": undefined,
          "query": undefined,
        },
        "responseObject": {
          "message": undefined,
          "status": undefined,
        },
        "timings": {
          "ms": 42,
        },
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
